### ISSUE TITLE
Issue with get_latest_hist_files

### ISSUE THREAD

**Issue by bartgol** [ESMCI/cime#4387]
I noticed that I _had_ to add `$` to the end of my `hist_file_extension`. Take the following
```
<comp_archive_spec compname="foo" compclass="atm">
  <hist_file_extension>my\.hist</hist_file_extension>
</comp_archive_spec>
```
and consider what CIME does for an ERS test:

1. run model for phase run1
2. get latest hist file, and copy it to $file.base
3. run model for phase run2
4. get latest hist file, and copy it to $file.rest
5. compare $file.base with $file.rest

Now, during step 4, CIME calls `get_latest_hist_files` _without_ suffix (cause it's fishing for the model output files). In that routine, [these](https://github.com/ESMCI/cime/blob/b6c527d28228821bbafaf7f8ca4f6f18cdf4566b/CIME/XML/archive_base.py#L85) lines are in charge of finding the "latest" file. However, they would think that `foo.my.hist.nc.base` comes _after_ `foo.my.hist.nc`, since it does come after in lexicographical order. Since the file does not end in `.nc`, [these lines](https://github.com/ESMCI/cime/blob/b6c527d28228821bbafaf7f8ca4f6f18cdf4566b/CIME/hist_utils.py#L81) would cause it to be skipped (in a way, thank god, or else we would get false positives due to self-comparison). Therefore, I would be left without a `.rest` file for step 5.

I think the assumption of in `get_latest_hist_files` is that the latest file will contain "the latest" timestamp, so lexicographical ordering should be fine. However, for ERS tests, this seems to be a broken assumption. Unless, the user regex _ends_ with `$`, but that forces the user to put a regex that includes _all_ the file content, and this is also not great (see issue #4382).

I think the fix should be easy: since `copy_histfiles` already _assumes_ that valid files end with `.nc`, perhaps we could do the same in `get_latest_hist_files`, and only consider files ending with `.nc` (therefore, skipping `blah.nc.base`).

Thoughts?

**Comment by jasonb5** [comment]
FAIL ERS.f09_g16_g.MALISIA.mappy_gnu (phase COMPARE_base_rest)
FAIL ERS.ne30pg2_r05_EC30to60E2r2.GPMPAS-JRA.mappy_gnu.mosart-rof_ocn_2way (phase COMPARE_base_rest)
FAIL ERS.r05_r05.IELM.mappy_gnu.elm-V2_ELM_MOSART_features (phase COMPARE_base_rest)
FAIL ERS_Ld5.T62_oQU120.CMPASO-NYF.mappy_gnu (phase COMPARE_base_rest)
FAIL ERS_Ld5.T62_oQU240.DTESTM.mappy_gnu (phase COMPARE_base_rest)
FAIL PEM_Ln5.T62_oQU240wLI.DTESTM.mappy_gnu (phase COMPARE_base_modpes)

These tests failed due to https://github.com/ESMCI/cime/pull/4389, which maybe be resolved once updating `get_latest_hist_files`.

**Comment by jasonb5** [comment]
I'll have a fix for this and a few other issues related to these functions this week.

**Comment by jasonb5** [comment]
This was fixed with #4452, CIME only considers `.nc` files when copying hist files for comparison during a test. This fix is available in the master branch of E3SM.
